FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/CustomFacetGatewayTest.php

```diff
@@ -0,0 +1,416 @@
+<?php
+
+use FinSearchUnified\Bundle\StoreFrontBundle\Gateway\Findologic\CustomFacetGateway;
+use FinSearchUnified\Constants;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\StoreFrontBundle\Service\ContextServiceInterface;
+use Shopware\Bundle\StoreFrontBundle\Struct\Search\CustomFacet;
+use Shopware\Components\Test\Plugin\TestCase;
+
+class CustomFacetGatewayTest extends TestCase
+{
+    public function tearDown()
+    {
+        parent::tearDown();
+        Shopware()->Container()->reset('front');
+        Shopware()->Container()->load('front');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testGetListMethodWhenShopSearchIsTriggered()
+    {
+        /** @var ContextServiceInterface $contextService */
+        $contextService = Shopware()->Container()->get('shopware_storefront.context_service');
+        $context = $contextService->getShopContext();
+
+        $mockOriginalService = $this->getMockBuilder(
+            '\Shopware\Bundle\StoreFrontBundle\Gateway\DBAL\CustomFacetGateway'
+        )
+            ->setMethods(['getList'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $mockOriginalService->expects($this->once())->method('getList');
+
+        $mockUrlBuilder = $this->getMockBuilder('\FinSearchUnified\Helper\UrlBuilder')->getMock();
+        $mockUrlBuilder->expects($this->never())->method('setCustomerGroup');
+        $mockUrlBuilder->expects($this->never())
+            ->method('buildCompleteFilterList');
+
+        $facetGateway = new CustomFacetGateway(
+            $mockOriginalService,
+            Shopware()->Container()->get('fin_search_unified.custom_listing_hydrator'),
+            $mockUrlBuilder
+        );
+
+        $facetGateway->getList([3], $context);
```

Comment:
yes apparently thats the first condition which the method checks for triggering the shop search if the `Request is null` so thats why it works because its triggering the shop search and uses the original service implementation